For debugging only.

This is a debug option used for memory stress testing.  Enter
a value in units of MiB to limit available hot plug memory.

---
 arch/powerpc/platforms/ps3/Kconfig |   17 +++++++++++++++++
 arch/powerpc/platforms/ps3/mm.c    |    8 ++++++++
 2 files changed, 25 insertions(+)

--- a/arch/powerpc/platforms/ps3/Kconfig
+++ b/arch/powerpc/platforms/ps3/Kconfig
@@ -162,4 +162,21 @@ config PS3_DEBUG_BOOT_MEM_LIMIT
 	  retail console reports 128 MiB of boot memory, this option
 	  will have no effect with a value of 128 or greater.
 
+config PS3_DEBUG_HOT_PLUG_MEM_LIMIT
+	depends on PPC_PS3
+	int  "PS3 Limit available hot plug memory"
+	range -1 1024
+	default -1
+	help
+	  This is a debug option used for memory stress testing.  Enter
+	  a value in units of MiB to limit available hot plug memory.
+
+	  If unsure, choose the default to disable this option.
+
+	  A negative value will disable this option.  A value of zero (0)
+	  will have maximum effect and the system will boot without hot
+	  plug memory.  Since the retail console reports about 128 MiB of
+	  hot plug memory, this option will have no effect with a value
+	  of 128 or greater.
+
 endmenu
--- a/arch/powerpc/platforms/ps3/mm.c
+++ b/arch/powerpc/platforms/ps3/mm.c
@@ -1227,6 +1227,14 @@ void __init ps3_mm_init(void)
 }
 #endif
 
+#if defined(CONFIG_PS3_DEBUG_HOT_PLUG_MEM_LIMIT) \
+	&& ((CONFIG_PS3_DEBUG_HOT_PLUG_MEM_LIMIT) >= 0)
+	map.total = min(map.total, map.rm.size
+		+ (CONFIG_PS3_DEBUG_HOT_PLUG_MEM_LIMIT) * 1024 * 1024);
+	pr_info("Limiting total RAM to 0x%llx (%llu MiB)\n", map.total,
+		map.total / (1024 * 1024));
+#endif
+
 	/* arrange to do this in ps3_mm_add_memory */
 	ps3_mm_region_create(&map.r1, map.total - map.rm.size);
 
